src/lib.rs

```diff
@@ -7,6 +7,125 @@
 
 #![allow(non_upper_case_globals, non_camel_case_types, non_snake_case, improper_ctypes)]
 
+//!
+//! This crate contains Rust bindings to the [SpiderMonkey Javascript engine][1]
+//! developed by Mozilla.
+//!
+//! These bindings are designed to be a fairly straightforward translation to the C++ API, while
+//! taking advantage of Rust's memory safety. For more about the Spidermonkey API, see the
+//! [API Reference][2] and the [User Guide][3] on MDN.
+//!
+//! [1]: https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey
+//! [2]: https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/JSAPI_reference
+//! [3]: https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/JSAPI_User_Guide
+//!
+//! # A minimal example
+//! Here is the code under "A minimal example" in the User Guide translated into Rust:
+//!
+//! ```rust
+//! use std::ffi::CStr;
```

Comment:
Per my comment in https://github.com/servo/rust-mozjs/pull/481#issuecomment-577588059, can we turn this into an actual example file rather than a docstring?